### ISSUE TITLE
ready for Emacs 29?

### ISSUE THREAD

**Issue by nhessler** [bbatsov/prelude#1403]
**Is your feature request related to a problem? Please describe.**

I'm curious if there is work going on to move prelude to be more compatible with the built-in treesitter, eglot, and use-package parts of emacs29

**Describe the solution you'd like**

maybe it could be a separate branch?

**Describe alternatives you've considered**

I'm open to anything and would be interested in helping if that's a concern.

**Additional context**

just wanted to get the conversation going. I'm super psyched for treesitter and eglot.

**Comment by rosingrind** [comment]
I'm new to `prelude`, using Emacs 29 and I've noticed at least 2 redundant (for 29+) dependencies:

- #1388 introduced `tree-sitter` dependency
- e83f905cdbd006542d488cdd302957710da6cdd1 introduced `use-package` dependency

AFAIK, both of these are included in 29+

**Comment by jasalt** [comment]
Long time Prelude user and currently attempting to move my personal config and best parts of Prelude over to [Emacs Bedrock](https://sr.ht/~ashton314/emacs-bedrock/) which is a new 29.1+ config starter kit.

**Comment by bbatsov** [comment]
> I'm curious if there is work going on to move prelude to be more compatible with the built-in treesitter, eglot, and use-package parts of emacs29

Yeah, that's definitely in the cards and help is welcome to make it happen faster. Of course, it's important to also support reasonably well at least Emacs 27 & 28, as Prelude normally aims to support the last 3 major Emacs releases.
